src/mo_frontend/bi_match.ml

```diff
@@ -128,18 +128,20 @@ let bi_match_subs scope_opt tbs subs typ_opt =
       | _ -> None
       )
     | Con (con1, ts1), t2 ->
-      (match Cons.kind con1, t2 with
-      | Def (tbs, t), _ -> (* TBR this may fail to terminate *)
-        bi_match_typ rel eq inst any (open_ ts1 t) t2
-      | Abs (tbs, t), _ when rel != eq ->
-        bi_match_typ rel eq inst any (open_ ts1 t) t2
-      | _ -> None
+      (match Cons.kind con1(*, t2*) with
+       | Def (tbs, t)(*, _*) -> (* TBR this may fail to terminate *)
+         bi_match_typ rel eq inst any (open_ ts1 t) t2
+       | Abs (tbs, t)(*, _*) when rel != eq ->
+         bi_match_typ rel eq inst any (open_ ts1 t) t2
+       | _ -> None
       )
     | t1, Con (con2, ts2) ->
       (match Cons.kind con2 with
-      | Def (tbs, t) -> (* TBR this may fail to terminate *)
-        bi_match_typ rel eq inst any t1 (open_ ts2 t)
-      | _ -> None
+       | Def (tbs, t) -> (* TBR this may fail to terminate *)
+         bi_match_typ rel eq inst any t1 (open_ ts2 t)
+       | Abs (tbs, t) when rel != eq ->
+         bi_match_typ rel eq inst any t1 (open_ ts2 t) (* Like above? ????? *)
```

Comment:
Similarly line 126
seems to be missing the mirrored case

Reply:
What about line 102?

Reply:
> @crusso do you agree that this `Abs` case is missing?

Actually, I don't agree. 

IIRC correctly, the original Abs case is promoting the (inflexible) variable T to its (upper) bound to check if the bound is less than the other type, i.e. T < t2 when T <: t1 and t1 <: t2 (here t1 is the upper bound on parameter T).

Since we only have upper, not lower bounds, the symmetric case doesn't arise. 

The suggested code would  actually be wrong, since we would infer that t1 <: T from T <: t2 and t1 <: t2, which isn't correct.

Reply:
> What about line 98, 102? And 227?

Maybe ask me over Zoom, I'm not sure what your asking about here.